src/main/kotlin/ecommerce/model/Product.kt

```diff
@@ -0,0 +1,41 @@
+package ecommerce.model
+
+import jakarta.persistence.CascadeType
+import jakarta.persistence.Column
+import jakarta.persistence.Entity
+import jakarta.persistence.FetchType
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.GenerationType
+import jakarta.persistence.Id
+import jakarta.persistence.OneToMany
+import jakarta.persistence.Table
+
+@Entity
+@Table(name = "products")
+class Product(
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    var id: Long? = null,
+    @Column(name = "name", nullable = false)
+    var name: String,
+    @Column(name = "price", nullable = false)
+    var price: Double,
+    @Column(name = "image_url", nullable = false)
+    var imageUrl: String,
+) {
+    @OneToMany(mappedBy = "product", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
+    var options: MutableList<Option> = mutableListOf()
+
+    override fun equals(other: Any?): Boolean {
+        if (this == other) return true
```

Comment:
```suggestion
        if (this === other) return true
```

Looks very similar, but acts in totally different way!
I would like you to tell me why! 😄

Reply:
==: is equal to the other
===: is same as the other -> same place in memory